pulsar-client-cpp/lib/OpSendMsg.h

```diff
@@ -25,9 +25,31 @@
 
 #include "TimeUtils.h"
 #include "MessageImpl.h"
+#include "ChunkMessageIdImpl.h"
+#include "ProducerImpl.h"
 
 namespace pulsar {
 
+class ProducerImpl;
+class ProducerChunkedMessageCtx {
+   public:
+    ProducerChunkedMessageCtx() = default;
+
+    ProducerChunkedMessageCtx(const MessageIdImplPtr firstChunkMessageId,
+                              const MessageIdImplPtr lastChunkMessageId)
+        : firstChunkMessageIdImplPtr_(firstChunkMessageId), lastChunkMessageIdImplPtr_(lastChunkMessageId) {}
+
+    MessageIdImplPtr getChunkMessageId() {
+        return std::make_shared<ChunkMessageIdImpl>(*firstChunkMessageIdImplPtr_, *lastChunkMessageIdImplPtr_)
+            ->getLastChunkMessageIdImpl();
+    }
```

Comment:
BTW, is this class used anywhere? I think we can remove the class definition. In Java, the same class is managed by Netty recycler to avoid frequent JVM heap memory allocation and frequent GC. In C++, there is no need to do that. I think the `MessageId` constructor that accepts two `MessageId` objects has already archived the same goal.

Reply:
Do you say the ProducerChunkedMessageCtx ？

Reply:
Yes.

Reply:
I mean the `getChunkMessageId` method is never used.

And it's very confused to make `firstChunkMessageIdImplPtr_` and `lastChunkMessageIdImplPtr_` private and access them directly in a friend class. It's a bad design.